docs/developer-guide.md

```diff
@@ -43,6 +43,20 @@ You can undeploy Katib v1beta1 manifests from a Kubernetes cluster as follows:
 make undeploy
 ```
 
+## Technical and style guide
+
+The following guidelines apply primarily to Katib, 
+but other projects like training-operator might also adhere to them. 
+
+When coding:
+
+- Follow [effective go](https://go.dev/doc/effective_go) guidelines.
```

Comment:
Maybe we should add that developer should use `gofmt` to format their code.

Reply:
It would be better to focus on mentioning only items that it is difficult to automatically detect in CI.
WDYT?

Reply:
It makes sense.
As an alternative, I would suggest using `make check`. WDYT?

https://github.com/kubeflow/katib/blob/46173463027e4fd2e604e25d7075b2b31a702049/Makefile#L31

Reply:
Sure, that even better, thanks!

Reply:
Done.